### ISSUE TITLE
Get Snapshot API: Listing all snapshots across all repositories does not work

### ISSUE THREAD

**Issue by spinscale** [elastic/elasticsearch#75997]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.13.4

**Description of the problem including expected versus actual behavior**:

The documentation for the [GET Snapshot API](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/get-snapshot-api.html) states for its path parameters:

```
GET /_snapshot/<repository>/<snapshot>

<repository>
(Required, string) Snapshot repository name used to limit the request.

To get information about all snapshot repositories registered in the cluster, omit this parameter or use * or _all.

<snapshot>
(Required, string) Comma-separated list of snapshot names to retrieve. Also accepts wildcards (*).

To get information about all snapshots in a registered repository, use a wildcard (*) or _all.
```

However trying to retrieve all snapshots from all repositories does not work.

**Steps to reproduce**:

```
GET _snapshot/_all/_all
```

returns

```json
{
  "error" : {
    "root_cause" : [
      {
        "type" : "repository_missing_exception",
        "reason" : "[_all] missing"
      }
    ],
    "type" : "repository_missing_exception",
    "reason" : "[_all] missing"
  },
  "status" : 404
}
```

Either this is a bug, or maybe the documentation needs to state that this is not supported.

**Comment by DaveCTurner** [comment]
TLDR this is a docs bug, you can only get from multiple repositories in ≥7.14.0, we shouldn't have backported this info to the docs for earlier versions.

Bit of a tricky one to disentangle however, we first introduced multi-repo support in `master` (#42090) but it was a breaking change to the response format so couldn't be backported to 7.x. But then we changed our minds about the new response format and adjusted `master` to keep using the old format (with slight tweaks). And then we backported a bunch of stuff in #74676 which included the multi-repo support.

The docs backport was #59238, looks like this went all the way back to 7.9.

**Comment by lockewritesdocs** [comment]
Closed by #76036
